### DISCUSSION TITLE
Vault with only 100 files doesn't get indexed. "An error occured when indexing your vault."

### DISCUSSION THREAD

**Discussion by sulth1** [logancyang/obsidian-copilot#341] [Q&A] [answered]
First, thank you so much for your work! This plugin is awesome.

My vault has 100 files and 130 000 tokens. Only a few files get indexed, like 2-3. I am not sure where the issue is. I have checked your video, tried the debug mode, tried the force indexed mode. 

I am still on the free 5$ OpenAI gives for new members. Could it be the reason?

**Comment by logancyang** [comment]
@sulth1 thanks for trying it out! Can you share a screenshot of your console?

**Reply by sulth1** [reply]
Thank you for the quick reply. Not really sure what to add here, there are so many errors. 

![image](https://github.com/logancyang/obsidian-copilot/assets/156309844/4f1a3018-edbe-49b8-a28e-ea3610dd6140)
 
![image](https://github.com/logancyang/obsidian-copilot/assets/156309844/71de9378-4e4c-499d-b071-bc09126ddd94)

According to that second screenshot, could it be that I just need to add a payment method, despite having the free 5$ available?

**Reply by logancyang** [reply, marked as answer]
429 means you are rate limited by OpenAI. They have a very low rate quota for free accounts nowadays. 

So if you want to use OpenAI you need to setup payment. Or just use ollama, keep in mind it's English only.

**Reply by sulth1** [reply]
Right. Does "to setup payment" mean paying X $ / month like a ChatGPT+ subscription, or just simply adding a payment method and pay for what I use? I see that the minimum amount to enter a card is 5$, so I would like to make sure before proceeding.

**Reply by logancyang** [reply]
It's pay for what you use, not a subscription
